### PR TITLE
added extra error checking for JWT plugin, see issue #1244

### PR DESCRIPTION
**Pull request by Tieske**

### CODE REVIEW

File: kong/plugins/jwt/handler.lua [hunk 1]
```diff
@@ -59,10 +59,15 @@ function JwtHandler:access(conf)
...
   end
+  if type(token) ~= "string" then
+    return responses.send_HTTP_UNAUTHORIZED("Unrecognizable token")
+  end
+  
 
   -- Decode token to find out who the consumer is
   local jwt, err = jwt_decoder:new(token)
   if err then
+    -- TODO: is this an internal server error? the token is invalid/cannot be parsed so isn't this a user error to be reported?
```
Comment (thefosk) [inline kong/plugins/jwt/handler.lua]: @thibaultCha ^

File: kong/plugins/jwt/handler.lua [hunk 2]
```diff
@@ -96,7 +101,12 @@ function JwtHandler:access(conf)
   if conf.secret_is_base64 then
     jwt_secret_value = jwt:b64_decode(jwt_secret_value)
   end
-
+  if not jwt_secret_value then
+    --TODO: shouldn't these values be checked when added? either secret or rsa_public_key is missing
+    -- or is not valid B64 encoded
```
Comment (thefosk) [inline kong/plugins/jwt/handler.lua]: @thibaultCha ^

File: kong/plugins/jwt/handler.lua [hunk 3]
```diff
@@ -16,14 +16,14 @@ JwtHandler.PRIORITY = 1000
...
--- @return token JWT token contained in request or nil
+-- @return token JWT token contained in request (can be a table) or nil
 -- @return err
 local function retrieve_token(request, conf)
   local uri_parameters = request.get_uri_args()
 
   for _, v in ipairs(conf.uri_param_names) do
     if uri_parameters[v] then
-      return uri_parameters[v]
+      return uri_parameters[v]  -- TODO: check multiple values, then this returns a table instead of a string
```
Comment (Tieske) [inline kong/plugins/jwt/handler.lua]: @thibaultCha this happens when there are mutiple tokens in the uri. What to do? error out, or accept the first and discard the others?

**Comment by thefosk** [comment]
We may be able to push this is 0.8.3 - @thibaultCha want to check the TODO comments?

**Comment by Tieske** [comment]
replaced by #1362

**Pull request closed without merging**